cabal-testsuite/PackageTests/UserConfig/cabal.test.hs

```diff
@@ -11,3 +11,7 @@ main = cabalTest $ do
     withEnv [("CABAL_CONFIG", Just conf2)] $ do
         cabal "user-config" ["init"]
         shouldExist conf2
+    cabalG ["--config-file", conf] "user-config" ["update", "-f", "-a", "extra-prog-path: foo", "-a", "extra-prog-path: bar"]
+    assertFileDoesContain conf "foo,bar"
+    cabalG ["--config-file", conf] "user-config" ["update", "-f", "-a", "extra-prog-path: foo, bar"]
```

Comment:
so if Windows user wants to write `extra-prog-path` they **should** use comma, semicolon is not accepted?

Reply:
Yes. This patch is to allow the parser to handle comma separated fields. Semicolon was never accepted anywhere. It "just happened" to work on windows (and actually still does because this patch won't change it) because it reads in "foo; bar" as a single value, and appends that single value to the path, but the syntax matches. I.e., it works by doing an deliberate "injection" attack :-).

Reply:
Can we change parser to recognise `;` as a separator too for that field. Than I think I'll be happy.

Reply:
... and the entry to `changelog`.

Reply:
do you want it as a separator for all fields or just to special case this one? i'd like to keep uniform syntax...

Reply:
As I mentioned on irc, changing the parser for filepath lists to handle `;` seems unnecessary to fix this -- it just extends the already convoluted parser code still further, adding special cases not only for certain lists (we already have like four distinct list types even without this) but also tokens.

Reply:
i.e. the docs already say
![image](https://user-images.githubusercontent.com/1125849/62160595-592d3a00-b2e2-11e9-9b8e-0892f785d14a.png)

so this patch as is doesn't change the grammar. it just fixes the fact that the docs claimed a feature that was only partially implemented.

Reply:
(and because, on unix and mac, the comma thing 'accidentally' worked, people didn't notice)